src/nm_mon_daemon.c

```diff
@@ -44,27 +44,31 @@ typedef struct nm_clean_data {
     { NM_MON_VMS_INIT, NULL, NULL, NULL, NM_THR_CTRL_INIT, NM_THR_CTRL_INIT }
 
 #if defined (NM_OS_LINUX)
-static void nm_mon_cleanup(int rc, void *arg)
+static nm_clean_data_t *clean_ptr = NULL;
+
+static void nm_mon_cleanup(void)
 {
-    nm_clean_data_t *data = arg;
     const nm_cfg_t *cfg = nm_cfg_get();
 
-    nm_debug("mon daemon exited: %d\n", rc);
+    nm_debug("mon daemon exited\n");
```

Comment:
We lost daemon rc code status?

Reply:
from `atexit(3)`:
```c
int atexit(void (*function)(void));
```

`atexit` function don't have any arguments

Reply:
I know this :) So why on_exit was used. Ok. We will return this code with other way.